website/database.py

```diff
@@ -728,16 +732,19 @@ def delete_class(self, Class):
     def resolve_class_link(self, link_id):
         return CLASSES.get({"link": link_id})
 
-    def get_username_invite(self, username):
-        return INVITATIONS.get({"username": username}) or None
+    def get_user_class_invite(self, username, class_id):
+        return INVITATIONS.get({"username#class_id": f"{username}#{class_id}"}) or None
+
+    def add_class_invite(self, key, data):
```

Comment:
```suggestion
    def add_class_invite(self, username, class_id, data):
```

If you do it like this, you can keep the concern that the `username` and `class_id` field have to be concatenated together using a `#` inside this class. That way, the rest of the code doesn't need to know that.

You could also grab those fields out of `data`, which is presumably a whole row, and then you don't need `key` at all:

```py
    def add_class_invite(self, data):
        data['username#class_id'] = data['username'] + '#' + data['class_id']
        # Put the entire record in the database
        INVITATIONS.put(data)
```